packages/framework/store/src/schema/base.ts

```diff
@@ -214,33 +250,18 @@ export class BlockModel<
     this.page = doc;
   }
 
-  get childMap() {
-    return this.children.reduce((map, child, index) => {
-      map.set(child.id, index);
-      return map;
-    }, new Map<string, number>());
-  }
-
   get children() {
-    const block = this.yBlock.get('sys:children') as Y.Array<string>;
-    if (!block) {
-      return [];
-    }
-
-    const children: BlockModel[] = [];
-    block.forEach(id => {
-      const child = this.doc.getBlockById(id);
-      if (!child) {
-        return;
-      }
-      children.push(child);
+    const computedChildren = computed(() => {
```

Comment:
it seems this is a memleak. A new watcher will be created on every `children` get.

Reply:
At first I use one signal that computed from `this._children`, however I found a problem that when y events emitted, the block instance haven't been created so the `if` in line 258 will get an `undefined`. I think we need to refactor the data flow to fix this problem. But I have no idea how to do that.

Reply:
two ideas:
1.

Reply:
Not really, once a block subscribe to the children, the future changes will notify the subscription. I think that's how the new `block-element` update the children without listener to `childrenUpdated` slot.

Reply:
Are you sure about that? getter like `get children` will only evaluate when it is being read, then it is done. There will be no way to make it an reactive attribute that allows it to be reactive to changes in the internal signals.
I guess it is being revaluated because this value is being read again that caused by some other effects.

Reply:
Not really sure, my guessing is that it will be subscribed when read because I subscribe it in `blockElement` with a [`SignalWatcher`](https://www.npmjs.com/package/@lit-labs/preact-signals#signalwatcher). I need to confirm the behavior.

Reply:
Based on our offline discussion. It's working in lit and still be a signal subscribed by the lit element. But the computed timing decides whether it can get the correct children list. It's a major discovery and we'll discuss more in dev committee.